examples/hotrod/pkg/tracing/http.go

```diff
@@ -18,45 +18,33 @@ package tracing
 import (
 	"context"
 	"encoding/json"
-	"errors"
-	"io"
 	"net/http"
 
-	"github.com/opentracing-contrib/go-stdlib/nethttp"
-	"github.com/opentracing/opentracing-go"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel/trace"
 )
 
 // HTTPClient wraps an http.Client with tracing instrumentation.
 type HTTPClient struct {
-	Tracer opentracing.Tracer
-	Client *http.Client
+	TracerProvider trace.TracerProvider
+	Client         *http.Client
 }
 
-// GetJSON executes HTTP GET against specified url and tried to parse
-// the response into out object.
-func (c *HTTPClient) GetJSON(ctx context.Context, endpoint string, url string, out interface{}) error {
-	req, err := http.NewRequest(http.MethodGet, url, nil)
-	if err != nil {
-		return err
+func NewHTTPClient(tp trace.TracerProvider) HTTPClient {
+	return HTTPClient{
+		TracerProvider: tp,
+		Client:         &http.Client{Transport: &otelhttp.Transport{}},
```

Comment:
you need to use NewTransport, not create the empty struct manually, and pass the Tracer as options.